core/src/main/scala/com/acervera/osm4scala/model/WayEntity.scala

```diff
@@ -25,15 +25,43 @@
 
 package com.acervera.osm4scala.model
 
-import org.openstreetmap.osmosis.osmbinary.osmformat.{StringTable, Way}
+import com.acervera.osm4scala.utilities.DecompressUtils
+import org.openstreetmap.osmosis.osmbinary.osmformat.{Info, StringTable, Way}
 
 /**
   * Entity that represent a OSM way as https://wiki.openstreetmap.org/wiki/Elements#Way and https://wiki.openstreetmap.org/wiki/Way describe
   */
-case class WayEntity(id: Long, nodes: Seq[Long], tags: Map[String, String]) extends OSMEntity {
+case class WayEntity(id: Long,
+                     nodes: Seq[Long],
+                     tags: Map[String, String],
+                     version: Option[Int],
+                     timestamp: Option[Long],
+                     changeset: Option[Long],
+                     uid: Option[Int],
+                     user: Option[String],
+                     visible: Option[Boolean]) extends OSMEntity {
 
   override val osmModel: OSMTypes.Value = OSMTypes.Way
 
+  def apply(id: Long,
```

Comment:
If all new parameters are optional, you can give them a default value of None.
Adding the default value, the user of the API will know that behaivor and `apply` will not be necessary.

Reply:
That sometimes problematic rasing overload prbblem which let me try this way.

Reply:
I saw lots of code written this way so I think it would be okay

Reply:
One reference is also here, https://github.com/databricks/scala-style-guide#default-parameter-values.

Reply:
I think that the link is related to Java interoperability. If you want to use it from Java you can not do it, and a lot of other things as well. :)

BTW, I did not know about the databricks style guide. Added to my Github stars.